Assets/MixedRealityToolkit.Tests/PlayModeTests/ButtonConfigHelperTests.cs

```diff
@@ -0,0 +1,118 @@
+// play mode tests in this check.
+
+using Microsoft.MixedReality.Toolkit.UI;
+using NUnit.Framework;
+using System.Collections;
+using UnityEditor;
+using UnityEngine;
+using UnityEngine.TestTools;
+
+namespace Microsoft.MixedReality.Toolkit.Tests
+{
+    public class ButtonConfigHelperTests : BasePlayModeTests
+    {
+        [UnityTest]
+        /// <summary>
+        /// Test adding a config helper to a game object and attempting to modify it.
+        /// </summary>
+        public IEnumerator TestAddButtonConfigHelperAtRuntime()
+        {
+            GameObject newGameObject = new GameObject("ButtonTest");
+            ButtonConfigHelper bch = newGameObject.AddComponent<ButtonConfigHelper>();
+
+            bch.MainLabelText = "Test Text";
+            bch.SeeItSayItLabelText = "Test Text";
+            bch.IconStyle = ButtonIconStyle.Char;
+            bch.IconStyle = ButtonIconStyle.None;
+            bch.IconStyle = ButtonIconStyle.Quad;
+            bch.IconStyle = ButtonIconStyle.Sprite;
+            bch.SeeItSayItLabelEnabled = false;
+            bch.SetCharIcon(0);
+            bch.SetQuadIcon(null);
+            bch.SetSpriteIcon(null);
+            bch.SetCharIconByName("EmptyIcon");
+            bch.SetQuadIconByName("EmptyIcon");
+            bch.SetSpriteIconByName("EmptyIcon");
+            bch.ForceRefresh();
+            bch.OnClick.AddListener(() => { Debug.Log("OnClick"); });
+            bch.IconSet = null;
+
+            yield break;
+        }
+
+        [UnityTest]
+        /// <summary>
+        /// Test creating a button icon set and inserting null elements.
+        /// </summary>
+        public IEnumerator TestBadlyConfiguredIconSet()
+        {
+            // Create a button icon set and insert arrays with null references
+            ButtonIconSet buttonIconSet = ScriptableObject.CreateInstance<ButtonIconSet>();
+
+            SerializedObject buttonIconSetObject = new SerializedObject(buttonIconSet);
+            SerializedProperty quadIconsProp = buttonIconSetObject.FindProperty("quadIcons");
+            SerializedProperty spriteIconsProp = buttonIconSetObject.FindProperty("spriteIcons");
+            SerializedProperty charIconsProp = buttonIconSetObject.FindProperty("charIcons");
+
+            quadIconsProp.InsertArrayElementAtIndex(0);
+            spriteIconsProp.InsertArrayElementAtIndex(0);
+            charIconsProp.InsertArrayElementAtIndex(0);
+            buttonIconSetObject.ApplyModifiedProperties();
+
+            // These calls should not fail even if we have null / empty elements.
+            Texture2D quadIcon;
+            Assert.IsFalse(buttonIconSet.TryGetQuadIcon("EmptyIcon", out quadIcon));
+
+            Sprite spritecon;
+            Assert.IsFalse(buttonIconSet.TryGetSpriteIcon("EmptyIcon", out spritecon));
+
+            uint charIcon;
+            Assert.IsFalse(buttonIconSet.TryGetCharIcon("EmptyIcon", out charIcon));
+            yield break;
+        }
+
```

Comment:
Per gilbdev's comment, it would be nice to add a test to verify that changing the text of a button at runtime changes the button's text.